File: services/shared/src/main/java/com/wedriveu/services/shared/utilities/Position.java
package com.wedriveu.services.shared.utilities;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * @author Marco Baldassarri
 * @author Michele
 * @since  17/07/2017
 */

public class Position {

    @JsonProperty
    private double latitude;
    @JsonProperty
    private double longitude;

    public Position() { }

    public Position(double latitude,
                    double longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getEuclideanDistance(Position position) {
        return Math.sqrt(Math.pow(this.latitude - position.getLatitude(), 2)
                + Math.pow(this.longitude - position.getLongitude(), 2));
    }

}

File: vehicle/src/main/scala/com/wedriveu/vehicle/entity/Position.scala
package com.wedriveu.vehicle.entity

import com.wedriveu.vehicle.shared.VehicleConstants

/**
  * @author Michele Donati on 17/07/2017.
  */
class Position(var latitude: Double, var longitude: Double) {

  def getDistanceInKm(position: Position): Double = {
    return VehicleConstants.earthRadiusInKm *
      Math.acos(Math.sin(this.latitude) *
        Math.sin(position.latitude) +
        Math.cos(this.latitude) *
          Math.cos(position.latitude) *
          Math.cos(this.longitude - position.longitude))
  }

}